defaults/main.yml

```diff
@@ -5,3 +5,18 @@ nrinfragent_service_state: "started"
 nrinfragent_config:
   license_key: YOUR_LICENSE_KEY
 nrinfragent_integrations: []
+nrinfragent_custom_version: "latest"
+nrinfragent_custom_installation:
+  "{{ nrinfragent_custom_bin_dir is defined
+  or nrinfragent_custom_mode is defined
+  or nrinfragent_custom_user is defined
+  or nrinfragent_custom_config_file is defined
+  or nrinfragent_custom_pid_file is defined
+  or nrinfragent_custom_agent_dir is defined
+  or nrinfragent_custom_plugin_dir is defined
+  or nrinfragent_custom_log_file is defined
+  or nrinfragent_custom_targz is defined
+  or nrinfragent_custom_version != ''}}"
+nrinfragent_custom_targz: "http://download.newrelic.com/infrastructure_agent/binaries/linux/amd64/newrelic-infra_linux_{{ nrinfragent_custom_version }}_amd64.tar.gz"
+nrinfragent_custom_tmpdir: "/tmp"
```

Comment:
If `tmp` is cleared, the file will be downloaded everytime ansible runs, even if the agent is already installed.

Reply:
I think it's ok, because I see this script running in two scenarios:
* Agent install, so the agent is downloaded for the first time and it is not present in any other directory.
* Agent update, so even if the agent tarball is there, it needs to be re-downloaded.

Reply:
The case I see is, the agent is already installed, `/tmp` is cleared by any reason, an ansible run is triggered, if everything is installed, why would I want to download the tar and uncompress it again. Is a minor thing, you can leave it like that.

Reply:
I think I'm missing something about the Ansible workflow. Why would you run Ansible many times on a single host, after it has been provisioned for the first time?